Runtime/Native/XBOX/NativeClient.cs

```diff
@@ -0,0 +1,140 @@
+        private static extern bool BacktraceAddFile(
+            [MarshalAs(UnmanagedType.LPWStr)] string name,
+            [MarshalAs(UnmanagedType.LPWStr)] string path
+        );
+
+        [DllImport("breakpad_xbox_simplified_abi_mt.dll")]
+        private static extern bool BacktraceSetUrl(
+            [MarshalAs(UnmanagedType.LPWStr)] string url
+        );
+
+        [DllImport("breakpad_xbox_simplified_abi_mt.dll")]
+        private static extern bool BacktraceBreakpadInitialized();
+
+        [DllImport("breakpad_xbox_simplified_abi_mt.dll")]
+        private static extern bool BacktraceInitializeBreakpad();
+
+        /// <summary>
+        /// Determine if the XBOX integration should be enabled
+        /// </summary>
+        private bool _enabled =
+#if UNITY_GAMECORE_XBOXSERIES && !UNITY_EDITOR
+            true;
+#else
+            false;
+#endif
+
+        public NativeClient(BacktraceConfiguration configuration, BacktraceBreadcrumbs breadcrumbs, IDictionary<string, string> clientAttributes, IEnumerable<string> attachments) : base(configuration, breadcrumbs)
+        {
+            if (!_enabled)
+            {
+                return;
+            }
+            AddScopedAttributes(clientAttributes);
+            HandleNativeCrashes(clientAttributes, attachments);
+            if (!configuration.ReportFilterType.HasFlag(ReportFilterType.Hang))
+            {
+                HandleAnr();
+            }
+        }
+
+        internal void AddScopedAttributes(IDictionary<string, string> attributes)
+        {
+            foreach (var attribute in attributes)
+            {
+                if (attribute.Key == null || attribute.Value == null)
+                {
+                    continue;
+                }
+
+                BacktraceAddAttribute(attribute.Key, attribute.Value);
+            }
+        }
+
+        private void HandleNativeCrashes(IDictionary<string, string> clientAttributes, IEnumerable<string> attachments)
+        {
+            var integrationDisabled = !_configuration.CaptureNativeCrashes || !_configuration.Enabled;
+            if (integrationDisabled)
+            {
+                return;
+            }
+
+            var minidumpUrl = new BacktraceCredentials(_configuration.GetValidServerUrl()).GetMinidumpSubmissionUrl().ToString();
+            foreach (var attachment in attachments)
+            {
+                var name = Path.GetFileName(attachment);
+                BacktraceAddFile(name, attachment);
+            }
+
+            BacktraceSetUrl(minidumpUrl);
+            CaptureNativeCrashes = BacktraceInitializeBreakpad();
+
+            if (!CaptureNativeCrashes)
+            {
+                Debug.LogWarning("Backtrace native integration status: Cannot initialize Crashpad client");
+                return;
+            }
+        }
+
+        public void GetAttributes(IDictionary<string, string> attributes)
+        {
+            throw new NotImplementedException();
```

Comment:
Is this expected?

Reply:
I don't see the value in implementing this method, but can do so. But it'll require additional implementation on the C-side and the hassle of marshalling the data through C-strings, for very dubious (IMO) gain.

Reply:
yeah but leaving not implemented exception is not something that our users want to see